coding-challenges.js

```diff
@@ -2,59 +2,162 @@
+  return arr.reduce((acc, val) => acc + (val ** 3), 0);
+  // initial value = 0
+  // intiVal + (2**3) = 8
+  // 8 becomes the accumulator (acc)
+  // acc + (3**3) = 35
+  // 35 becomes the acc
+  // acc + (4**3) = 99
+}
+
+console.log(sumCubed(cubeAndSum1));
+console.log(sumCubed(cubeAndSum2));
 
 // --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
-
-
-
-
 var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
 var nums2 = [109, 5, 9, -59, 8, 24]
 // Expected output: [-59, 109]
 
+const minMax = arr => {
+  return [Math.min(...arr), Math.max(...arr)];
+}
 
-// --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
-
-
+console.log(minMax(nums1));
+console.log(minMax(nums2));
 
 
 
+// --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
 var testString1 = "albatross"
 // Expected output: "aLbAtRoSs"
 var testString2 = "jabberwocky"
 // Expected output: "jAbBeRwOcKy"
 
 
-// --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
+// VERSION 1.0
+const coolLettersV1 = str => {
+  // converts string to array
+  let newStr = str.split("");
+  // loops through array toUpperCase in odd indexes
+  for (let i = 1; i < newStr.length; i += 2) {
+    newStr[i] = newStr[i].toUpperCase();
+  }
+  // join array back into a single string
+  return newStr.join("");
+}
 
 
+// VERSION 2.0
+const coolLettersV2 = str => {
+  // split string into an array of letters
+  // map through letter with the value and index as arguments
+  // use ternary to test if index is an odd num
+    // if yes, toUpperCase that letter
+    // if not, return normally
+  return str.split("")
+  .map((char, i) => i % 2 !== 0 ? char.toUpperCase() : char)
+  .join("");
+}
 
 
 
+
+
+// --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
 var arr1 = [3, 7, 10, 5, 4, 3, 3]
 var arr2 = [7, 8, 2, 3, 1, 5, 4]
+
+// VERSION 1.0 USING A FOR LOOP
+const noDupsV1 = (arr1, arr2) => {
+  // does arr2 includes a num from arr1?
+  let newArr = [];
+  let tempArr = [...arr1,...arr2];
+
+  for (let i = 0; i < tempArr.length; i++) {
+    if (newArr.indexOf(tempArr[i]) === -1) {
+      newArr.push(tempArr[i]);
+    }
+  }
+
+  return newArr;
+}
+
+// VERSION 2.0 USING FILTER METHOD
+const noDupsV2 = (arr1, arr2) => {
+  let newArr = [];
+  let tempArr = [...arr1,...arr2];
+
+  // Just another way of writing the exact same thing as line 154
+  // tempArr.filter(num => {
+  //   if (newArr.indexOf(num) === -1) {
+  //     newArr.push(num);
+  //   }
+  // });
+
+  return newArr = tempArr.filter(num => newArr.indexOf(num) === -1);
+}
+
```

Comment:
i love that you have multiple approaches for your solutions. that's a very valuable habit to have in interviews. employers what to see all the different way you go about solving a problem and re-factoring